### PR TITLE
UI Text - Add support, docs and tests for dynamic properties

### LINKED ISSUE #1176: Dynamic Properties: UI Text
Using [Dynamic Properties Best Practices](https://dashboard.flowfuse.com/contributing/widgets/core-widgets.html#dynamic-properties), implement:

- Label
- Layout
- Font
- FontSize
- Color

as dynamic properties.

### PR DESCRIPTION
**Pull request by joepavitt**
## Description

- Adds dynamic support for properties detailed in #1176 
- Adds E2E coverage for all dynamic properties introduced
- Adds supporting documentation in-Editor and in the online documentation

## Related Issue(s)

Closes #1176

### CODE REVIEW

**Review by gayanSandamal — APPROVED** [review]
LGTM

File: nodes/widgets/ui_text.js [hunk 1]
```diff
@@ -19,80 +21,40 @@ module.exports = function (RED) {
...
+                }
+                if (typeof updates.fontSize !== 'undefined') {
+                    // dynamically set "label" property
+                    statestore.set(group.getBase(), node, msg, 'fontSize', updates.fontSize)
+                }
+                if (typeof updates.color !== 'undefined') {
+                    // dynamically set "label" property
+                    statestore.set(group.getBase(), node, msg, 'color', updates.color)
+                }
+            }
```
Comment (Steve-Mcl) [inline nodes/widgets/ui_text.js:47]: Should this not be using the new helper functions?

File: nodes/widgets/locales/en-US/ui_text.html [hunk 2]
```diff
@@ -26,7 +26,23 @@ <h3>HTML Content</h3>
     <h3>Dynamic Properties (Inputs)</h3>
     <p>Any of the following can be appended to a <code>msg.</code> in order to override or set properties on this node at runtime.</p>
```
Comment (Steve-Mcl) [inline nodes/widgets/locales/en-US/ui_text.html:27]: This should have been updated for `msg.ui_update`

Comment (joepavitt) [reply nodes/widgets/locales/en-US/ui_text.html:27]: Helpers were only added client-side

Comment (joepavitt) [reply nodes/widgets/locales/en-US/ui_text.html:27]: Right you are

**Comment by Steve-Mcl** [comment]
I am playing on main branch where this code is live and I have a question about behaviour when dynamic props are applied.

As it stands, doing a ui_update to `class` or `label` (and likely others) resets the text of the `ui-text`. Is this the intended behaviour?

chrome_NmeIGnHqOl

```
[{"id":"3f4a2740a6e91c95","type":"ui-text","z":"dcc15970006a41ac","group":"9b597a79245141ec","order":2,"width":"3","height":"1","name":"","label":"left","format":"{{msg.payload}}","layout":"row-spread","style":false,"font":"","fontSize":16,"color":"#717171","className":"hideText","x":410,"y":420,"wires":[]},{"id":"01ddda3a3b9eb8b3","type":"ui-text","z":"dcc15970006a41ac","group":"9b597a79245141ec","order":1,"width":"3","height":"1","name":"","label":"right","format":"{{msg.payload}}","layout":"row-spread","style":false,"font":"","fontSize":16,"color":"#717171","className":"","x":410,"y":460,"wires":[]},{"id":"781f36df75997293","type":"inject","z":"dcc15970006a41ac","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":true,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":170,"y":480,"wires":[["83204da5a34ac7a6"]]},{"id":"809d7fc2ea9e1053","type":"ui-button","z":"dcc15970006a41ac","group":"9b597a79245141ec","name":"","label":"button","order":3,"width":"3","height":"1","emulateClick":false,"tooltip":"","color":"","bgcolor":"","className":"","icon":"","iconPosition":"left","payload":"","payloadType":"str","topic":"topic","topicType":"msg","x":410,"y":520,"wires":[[]]},{"id":"ebb6bcf4373e717a","type":"ui-template","z":"dcc15970006a41ac","group":"","page":"","ui":"22ea43815413e748","name":"","order":0,"width":0,"height":0,"head":"","format":"\n.hideText {\n    display: none !important;\n}\n\n.nrdb-ui-widget.nrdb-ui-text.showText {\n    display: flex !important;\n}\n","storeOutMessages":true,"passthru":true,"resendOnRefresh":true,"templateScope":"site:style","className":"","x":420,"y":380,"wires":[[]]},{"id":"321042e1a918adf2","type":"inject","z":"dcc15970006a41ac","name":"class hideText","props":[{"p":"class","v":"hideText","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":170,"y":380,"wires":[["83204da5a34ac7a6"]]},{"id":"789871d57db2a0e7","type":"inject","z":"dcc15970006a41ac","name":"class showText","props":[{"p":"class","v":"showText","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":180,"y":420,"wires":[["83204da5a34ac7a6"]]},{"id":"2ffc6c2d9e9203c8","type":"inject","z":"dcc15970006a41ac","name":"ui_update.label ggg","props":[{"p":"ui_update.label","v":"ggg","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":190,"y":280,"wires":[["83204da5a34ac7a6"]]},{"id":"042bb976a28f4343","type":"inject","z":"dcc15970006a41ac","name":"ui_update.label hhh","props":[{"p":"ui_update.label","v":"hhh","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":190,"y":320,"wires":[["83204da5a34ac7a6"]]},{"id":"83204da5a34ac7a6","type":"junction","z":"dcc15970006a41ac","x":320,"y":440,"wires":[["01ddda3a3b9eb8b3","3f4a2740a6e91c95"]]},{"id":"9b597a79245141ec","type":"ui-group","name":"Test-Group","page":"a12cc971e8c42860","width":"6","height":"1","order":1,"showTitle":true,"className":"","visible":"true","disabled":"false"},{"id":"22ea43815413e748","type":"ui-base","name":"base","path":"/dashboard","includeClientData":true,"acceptsClientConfig":["ui-notification","ui-control"],"showPathInSidebar":false,"navigationStyle":"default","titleBarStyle":"default"},{"id":"a12cc971e8c42860","type":"ui-page","name":"Test-Page","ui":"22ea43815413e748","path":"/page25","icon":"home","layout":"grid","theme":"c2ff5ba1f92a0f0e","order":5,"className":"","visible":"true","disabled":"false"},{"id":"c2ff5ba1f92a0f0e","type":"ui-theme","name":"Default","colors":{"surface":"#ffffff","primary":"#0094ce","bgPage":"#eeeeee","groupBg":"#ffffff","groupOutline":"#cccccc"},"sizes":{"pagePadding":"12px","groupGap":"12px","groupBorderRadius":"4px","widgetGap":"12px"}}]
```

**Comment by bartbutenaers** [comment]
@Steve-Mcl 
Might that be related to the TODO [here](https://github.com/FlowFuse/node-red-dashboard/blob/fff158d200973f68d6d6d23b952434efe9058707/ui/src/widgets/data-tracker.mjs#L82) perhaps?  Because the default onInput seems to store every message   I think it should store only messages that have a payload.  Otherwise when you inject an msg without payload (to set some property dynamically) that empty payload will be used as value.  I think...

**Pull request merged**